include/class.list.php

```diff
@@ -229,19 +229,27 @@ function getItems($limit=false, $offset=false) {
 
 
 
-    function getItem($val) {
+    function getItem($val, $extra=false) {
+
+        $items = DynamicListItem::objects()->filter(
+                array('list_id' => $this->getId()));
 
-        $criteria = array('list_id' => $this->getId());
         if (is_int($val))
-            $criteria['id'] = $val;
+            $items->filter(array('id' => $val));
+        elseif ($extra)
+            $items->filter(array('extra' => $val));
         else
-            $criteria['value'] = $val;
+            $items->filter(array('value' => $val));
+
 
-         return DynamicListItem::lookup($criteria);
+        return $items->one();
```

Comment:
This will throw an exception in `develop-next`. Seems like we should catch it here